### ISSUE TITLE
Errors from Pouch are swallowed

### ISSUE THREAD

**Issue by dave-irvine** [ArnoSaine/react-pouchdb#7]
Hi!

Great work on this library, but I think I found an issue.

If you pass some parameter to `useFind` that causes Pouch to throw an error (my problem was I asked to sort on a column that has no index), this error is swallowed and never shows up in the console. I had to step deep into the code stack to eventually find the error:

```
Unhandled promise rejection Error: "Cannot sort on field(s) "firstName" when using the default index"
    validateSort index-browser.es.js:501
    1 index-browser.es.js:1216
    run es.promise.js:97
    notify es.promise.js:114
    flush microtask.js:25
    argsArray index.js:18
    _callee$ useFind.js:55
    tryCatch runtime.js:45
    invoke runtime.js:271
    method runtime.js:97
    asyncGeneratorStep asyncToGenerator.js:3
    _next asyncToGenerator.js:25
    run es.promise.js:97
    notify es.promise.js:114
    flush microtask.js:25
```

The result of this is that `useFind()` returns `undefined` and you never know there was a problem. Now that I know the problem I can sort out my index, but the issue I am reporting is that react-pouchdb swallows errors from Pouch.

**Comment by ArnoSaine** [comment]
Hi!

Thank you for the support and submitting this issue.

It's a real issue. I think errors could be thrown and captured using an error boundary. Fixing this will take a few moments, because I'll also rewrite and simplify the subscription logic.

**Comment by gajus** [comment]
Just started using this library and quickly ran into the same issue.

**Comment by ArnoSaine** [comment]
@gajus, nice to get new users! Could you try the new beta release, `npm i react-pouchdb@next`? This should be fixed in that release.

Change log:
https://github.com/ArnoSaine/react-pouchdb/blob/next/CHANGELOG.md#200-beta0

Docs:
https://www.npmjs.com/package/react-pouchdb/v/2.0.0-beta.1

**Comment by gajus** [comment]
Sorry, stripped PouchDB from client and ended up writing an adhoc "database" using https://github.com/paularmstrong/normalizr. It is better suited for an esoteric task that I have.

**Comment by ArnoSaine** [comment]
Ok, no worries 🙂

**Comment by ArnoSaine** [comment]
Fixed in 2.0.0 🎉
